masp_primitives/src/transaction/components/amount.rs

```diff
@@ -78,30 +78,64 @@ where
 impl<Unit, Value> ValueSum<Unit, Value>
 where
     Unit: Hash + Ord + BorshSerialize + BorshDeserialize + Clone,
-    Value: BorshSerialize + BorshDeserialize + PartialEq + Eq + Copy + Default + PartialOrd,
+    Value: BorshSerialize + BorshDeserialize + PartialEq + Eq + Copy + Zero + PartialOrd,
 {
     /// Creates a non-negative ValueSum from a Value.
     pub fn from_nonnegative(atype: Unit, amount: Value) -> Result<Self, ()> {
-        if amount == Value::default() {
+        if amount.is_zero() {
             Ok(Self::zero())
-        } else if Value::default() <= amount {
+        } else if Value::zero() <= amount {
             let mut ret = BTreeMap::new();
             ret.insert(atype, amount);
             Ok(ValueSum(ret))
         } else {
             Err(())
         }
     }
+
+    /// Compute the infimum of two ValueSums
+    pub fn inf(&self, rhs: &Self) -> Self {
+        let mut comps = BTreeMap::new();
+        for (atype, rhs_amount) in rhs.components() {
+            // rhs_amount is positive by definition
```

Comment:
Does this guarantee comes from the fact that notes always carry positive amounts? Cause `ValueSum`s could actually carry any type

Reply:
Ah, you're correct. I made a bad assumption. I think the issue is now fixed.